File: javascript/src/warmup/isPangram.js
export function isPangram(sentence) {
    // time: O(n)
    // space: O(1)
    let chars = new Set();

    for (const char of sentence.toLowerCase()) {
        if (char.match(/[a-z/i]/)) {
            chars.add(char);
        }
    }

    return chars.size === 26;
}

File: rust/src/lib/warmup/is_pangram.rs
use std::collections::HashSet;

fn is_pangram(sentence: &str) -> bool {
    // time: O(n)
    // space: O(1)
    let set: HashSet<char> = sentence
        .to_lowercase()
        .chars()
        .filter(|&c| c.is_ascii_alphabetic())
        .collect();

    set.len() == 26
}

#[cfg(test)]
pub mod test {
    use super::*;

    #[test]
    fn test_is_pangram() {
        let test_cases = [
            ("This is not a pangram", false),
            ("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ", true),
        ];

        for (sentence, expected) in test_cases {
            assert_eq!(is_pangram(sentence), expected);
        }
    }
}